fivaldi/ctvero-rocnich-obdobi
# Add contest(s) page, update CSS/JS deps, refactor code incl. fixes and minor enhancements

- Add contest(s) page incl. interactive map (Mapbox/Leaflet)
- Contest calendar contains alarms and submission deadline event (reminder)
- Set page titles from controllers
- Rename DiaryController::listContestDiaries ->
  DiaryController::getContestDiaries
- Translate URL path components/params from CS to EN (existing CS routes
  will redirect to EN routes)
- Use URL #scroll hash param to scroll pages on some occasions (form submission
  continuation, jump to a contest page from list of contests, jump to contest
  results etc.)
- Make App\Http\Utilities central place for helpers
- Contest DB model has new column: map_marker_color (see contest page)
- Diary DB model has new columns: qth_locator_lon, qth_locator_lat
- Make Illuminate\Support\Str and App\Http\Utilities accessible easily
  as facades
- Add rexxars/html-validator (used for XHTML validation in unit tests)
  to composer.json
- Move league/flysystem (Storage) from dev to prod in composer.json
- Make Mapbox access token configurable using env('CTVERO_MAPBOX_ACCESS_TOKEN')
- Add QthLongitudeLatitude migration (and set locator lon/lat for all diaries)
- Add CategoryMapMarkerColor migration (and set colors for all categories)
- Add Mapbox access token to .env.gpg
- Update composer-setup.php hash
- Do not install DEV deps during deployment
- Upgrade Bootstrap from v4.0.0-beta.2 to 4.6.0 and use CDN (and delete
  static files incl. deps: jQuery, Popper)
- Add overrides to custom.css (minor tweaks, responsivity etc.)
- Ugrade Font Awesome from 4.5.0 to 4.7.0
- Add new images to Slick carousel with big thanks to @Apache Znojomo,
  @Salamandr Velká Bystřice and @Oto Štěpánkovice
- Remove database/migrations/.gitkeep and resources/views/.gitkeep
- Add public/static/js/.gitkeep
- Upgrade Slick to 1.8.1 and remove unnecessary files
- Add component for results table (and improve responsivity)
- Set error page title according to the error message
- Index template: use list group for contests, improve contact form
  responsivity
- Remove unused http://www.w3schools.com/lib/w3data.js
- Fix automated scrolling behavior after whole document is loaded (see
  #scroll above)
- Use lazy loading for Slick carousel
- Submission menu item displays only when needed (and shows days/hours/minutes
  till deadline)
- Fix XHTML and JS in templates
- Fix alert in news (DE)
- Add unit tests: cs→en redirects, contest(s) pages, XHTML validation of
  rendered pages

/cc @Podly27
